scripts/override-start.js

```diff
@@ -24,8 +24,28 @@ const overrideWebpackConfig = ( config, devServer ) => {
 	// Replace the react-dev-utils webpackHotDevClient with a version patched to
 	// correctly detect the dev server host & port for socket requests.
 	const hotClient = require.resolve( 'react-dev-utils/webpackHotDevClient' );
-	const hotClientIndex = config.entry.indexOf( hotClient );
-	config.entry.splice(hotClientIndex, 1, require.resolve( '../overrides/webpackHotDevClient' ) );
+
+	if ( config.entry instanceof Array ) {
```

Comment:
Just a hint that the _modern_ way of doing this is `Array.isArray( config.entry )`. While irrelevant here, this is also faster with modern browsers.